### PR TITLE
Removing Connection.lookup in storage.

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

File: gcloud/storage/connection.py [hunk 1]
```diff
@@ -169,7 +169,11 @@ def __iter__(self):
         return iter(_BucketIterator(connection=self))
 
     def __contains__(self, bucket_name):
-        return self.lookup(bucket_name) is not None
+        try:
+            self.get_bucket(bucket_name)
+            return True
```
Comment (tseaver) [inline gcloud/storage/connection.py]: Would it be cleaner to move the `return` in an `else:` block?

**Comment by tseaver** [comment]
Is there motivation for dropping `lookup`?

File: gcloud/storage/connection.py (hunk 1 above)
Comment (dhermes) [inline gcloud/storage/connection.py]: I don't have a sense for what is or isn't clean in this instance. Happy to change.

**Comment by dhermes** [comment]
Motivation for dropping `lookup` is that 

1. It's essentially the same as `get_bucket`
1. `Connection` methods should just be making API requests and leaving more user-friendly behavior to the other classes

**Comment by tseaver** [comment]
I'm not sure I buy dropping `lookup()`:  like `dict.get()`, it makes for smoother flow in calling code than always having to `try: ... except: ...` around KeyErrors/NotFound.

How likely is it that we will have applications dealing with multiple, non-default buckets?  The more likely such cases, the less I'd be inclined to drop `lookup()`.

**Comment by dhermes** [comment]
@tseaver Rebased after #580.

I agree about smoother flow in calling code, but am working under the idea that `Connection` should not be the surface that users deal with (hence not in calling code).

The idea is to put the user-friendly type methods in a module like `datastore.api` and surface in the same way (like `datastore.get`, `datastore.put` and `datastore.delete`).

WDYT?

**Comment by tseaver** [comment]
So, do you see us adding API functions like `storage.lookup()`:

```python
def lookup(bucket_name, connection=None):
    """Look up a bucket by name.

    :type bucket_name: string
    :param bucket_name: the name of the bucket being looked up.

    :type connection: :class:`gcloud.storage.connection.Connection`, or None.
    :param connection: the connection to use.  If None, use the connection
                       inferred from the environment.

    :rtype: :class:`gcloud.storage.bucket.Bucket`, or None
    ;returns: the bucket, if it exists, or None if not.
    """
```

**Comment by dhermes** [comment]
I do, with a fallback to the implicit connection. As with `datastore`, we should put it out of view for most users to reduce complexity / surface area.

**Comment by tseaver** [comment]
Do you mean put the `storage.lookup` API out of sight?

**Comment by dhermes** [comment]
Nope, I meant putting the `Connection` out of sight.

**Comment by tseaver** [comment]
"Hiding" the `Connection` class SGTM.

**Comment by dhermes** [comment]
Great. I'll follow this up with a creation of the `storage.api` module or you want me to do it here before an LGTM?

**Comment by tseaver** [comment]
LGTM

**Pull request merged**